go/service/chat_local.go

```diff
@@ -130,17 +210,76 @@ getinbox:
 		}
 	}
 
-	id, err := h.NewConversationLocal(ctx, chat1.ConversationIDTriple{
-		Tlfid:     tlfID,
-		TopicType: arg.TopicType,
-		// TopicID filled by server?
-	})
-	if err != nil {
-		return ids, err
+	return conversations, nil
+}
+
+// getConversationInfo locates the conversation by using id, and returns with
+// all fields filled in conversationInfo, along with a ConversationIDTriple
+//
+// TODO: cache
+func (h *chatLocalHandler) getConversationInfo(ctx context.Context, id chat1.ConversationID) (conversationInfo keybase1.ConversationInfoLocal, triple chat1.ConversationIDTriple, err error) {
+	ipagination := &chat1.Pagination{Num: 20}
+getinbox:
+	for i := 0; i < 10000; /* in case we have a server bug */ i++ {
+		iview, err := h.GetInboxLocal(ctx, ipagination)
+		if err != nil {
+			return conversationInfo, triple, err
+		}
+
+		for _, conversation := range iview.Conversations {
+			if id == conversation.Metadata.ConversationID {
+				triple = conversation.Metadata.IdTriple
+				conversationInfo.Id = id
+				conversationInfo.TopicType = triple.TopicType
+
+				if len(conversation.MaxHeaders) == 0 {
+					return conversationInfo, triple, errors.New("empty conversation found")
```

Comment:
While I agree this is an attack to worry about, I'm not convinced this is the only or best defense.  Why can GetInbox or whatever points to the conversation return a <TLFName, TopicType, TopicID> triple, and then I call `GetCryptKeys` on the TLFName?

I can stash a local TLFId <-> TLFName mapping on the local device btw, but would need to do the call on the TLFName at least once per device.